sensu/gohandler.go

```diff
@@ -1,219 +1,52 @@
-	}
-
-	if err = validateEvent(sensuEvent); err != nil {
-		return err
-	}
-
-	goHandler.sensuEvent = sensuEvent
-	return nil
-}
-
-func validateEvent(event *types.Event) error {
-	if event.Timestamp <= 0 {
-		return errors.New("timestamp is missing or must be greater than zero")
-	}
-
-	if event.Entity == nil {
-		return errors.New("entity is missing from event")
-	}
-
-	if !event.HasCheck() {
-		return errors.New("check is missing from event")
-	}
-
-	if err := event.Entity.Validate(); err != nil {
-		return err
-	}
-
-	if err := event.Check.Validate(); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func configurationOverrides(config *HandlerConfig, options []*HandlerConfigOption, event *types.Event) error {
-	if config.Keyspace == "" {
-		return nil
-	}
-	for _, opt := range options {
-		if len(opt.Path) > 0 {
-			// compile the Annotation keyspace to look for configuration overrides
-			k := path.Join(config.Keyspace, opt.Path)
-			switch {
-			case len(event.Check.Annotations[k]) > 0:
-				err := setOptionValue(opt, event.Check.Annotations[k])
-				if err != nil {
-					return err
-				}
-				log.Printf("Overriding default handler configuration with value of \"Check.Annotations.%s\" (\"%s\")\n", k, event.Check.Annotations[k])
-			case len(event.Entity.Annotations[k]) > 0:
-				err := setOptionValue(opt, event.Entity.Annotations[k])
-				if err != nil {
-					return err
-				}
-				log.Printf("Overriding default handler configuration with value of \"Check.Annotations.%s\" (\"%s\")\n", k, event.Entity.Annotations[k])
-			}
-		}
-	}
-	return nil
-}
-
-func setOptionValue(option *HandlerConfigOption, valueStr string) error {
-	switch option.Value.(type) {
-	case *string:
-		strOptionValue, ok := option.Value.(*string)
-		if ok {
-			*strOptionValue = valueStr
-		}
-	case *uint64:
-		uint64OptionPtrValue, ok := option.Value.(*uint64)
-		if ok {
-			parsedValue, err := strconv.ParseUint(valueStr, 10, 64)
-			if err != nil {
-				return fmt.Errorf("Error parsing %s into a uint64 for option %s", valueStr, option.Argument)
-			}
-			*uint64OptionPtrValue = parsedValue
-		}
-	case *bool:
-		boolOptionPtrValue, ok := option.Value.(*bool)
-		if ok {
-			parsedValue, err := strconv.ParseBool(valueStr)
-			if err != nil {
-				return fmt.Errorf("Error parsing %s into a bool for option %s", valueStr, option.Argument)
-			}
-			*boolOptionPtrValue = parsedValue
-		}
-	}
-	return nil
-}
-
-// Intentionally does nothing since we're only using cobra to read the command line arguments
-func (goHandler *GoHandler) cobraExecute(_ []string) error {
-	// Read Sensu event
-	err := goHandler.readSensuEvent()
-	if err != nil {
-		return err
-	}
-
-	// Override the configuration with the event information
-	err = configurationOverrides(goHandler.config, goHandler.options, goHandler.sensuEvent)
-	if err != nil {
-		return err
-	}
-
+// Executes the handler's workflow
+func (goHandler *GoHandler) goHandlerWorkflow(_ []string) (int, error) {
```

Comment:
Is the `int` returned here the exit status of the function?

Reply:
It's actually the exit status of the handler. GoPlugin.Execute will use this value for the os.Exit call. Since different types of plugins (handler, checks, mutators, etc...) have different specifications about what that value should be I felt it was best for each type of plugin implement their own logic to determine the value.